cnf-tests/testsuites/e2esuite/dpdk/dpdk.go

```diff
@@ -946,55 +1044,66 @@ func CreateSriovNetwork(sriovDevice *sriovv1.InterfaceExt, sriovNetworkName stri
 	}, time.Minute, 5*time.Second).ShouldNot(HaveOccurred())
 }
 
-func createDPDKWorkload(nodeSelector map[string]string, dpdkResourceName, testpmdCommand string, runningTime int, isServer bool) (*corev1.Pod, error) {
+func dpdkWorkloadCommand(dpdkResourceName, testpmdCommand string, runningTime int) string {
+	return fmt.Sprintf(`set -ex
+	export CPU=$(cat /sys/fs/cgroup/cpuset/cpuset.cpus)
+	echo ${CPU}
+	echo ${PCIDEVICE_OPENSHIFT_IO_%s}
+	
+	cat <<EOF >test.sh
+	spawn %s
+	set timeout 10000
+	expect "testpmd>"
+	send -- "port stop 0\r"
+	expect "testpmd>"
+	send -- "port detach 0\r"
+	expect "testpmd>"
+	send -- "port attach ${PCIDEVICE_OPENSHIFT_IO_%s}\r"
+	expect "testpmd>"
+	send -- "port start 0\r"
+	expect "testpmd>"
+	send -- "start\r"
+	expect "testpmd>"
+	sleep %d
+	send -- "stop\r"
+	expect "testpmd>"
+	send -- "quit\r"
+	expect eof
+	EOF
+	
+	expect -f test.sh
+	
+	sleep INF
+	`, dpdkResourceName, testpmdCommand, dpdkResourceName, runningTime)
+}
+
+func createDPDKWorkload(nodeSelector map[string]string, command string, isServer bool, additionalCapabilities []corev1.Capability) (*corev1.Pod, error) {
 	resources := map[corev1.ResourceName]resource.Quantity{
 		corev1.ResourceName("hugepages-1Gi"): resource.MustParse("2Gi"),
 		corev1.ResourceMemory:                resource.MustParse("1Gi"),
 		corev1.ResourceCPU:                   resource.MustParse("4"),
 	}
+
+	capabilities := []corev1.Capability{"IPC_LOCK", "SYS_RESOURCE", "NET_RAW"}
+	if additionalCapabilities != nil {
+		capabilities = append(capabilities, additionalCapabilities...)
+	}
+
 	container := corev1.Container{
 		Name:  "dpdk",
 		Image: images.For(images.Dpdk),
 		Command: []string{
 			"/bin/bash",
 			"-c",
-			fmt.Sprintf(`set -ex
-export CPU=$(cat /sys/fs/cgroup/cpuset/cpuset.cpus)
-echo ${CPU}
-echo ${PCIDEVICE_OPENSHIFT_IO_%s}
-
-cat <<EOF >test.sh
-spawn %s
-set timeout 10000
-expect "testpmd>"
-send -- "port stop 0\r"
-expect "testpmd>"
-send -- "port detach 0\r"
-expect "testpmd>"
-send -- "port attach ${PCIDEVICE_OPENSHIFT_IO_%s}\r"
-expect "testpmd>"
-send -- "port start 0\r"
-expect "testpmd>"
-send -- "start\r"
-expect "testpmd>"
-sleep %d
-send -- "stop\r"
-expect "testpmd>"
-send -- "quit\r"
-expect eof
-EOF
-
-expect -f test.sh
-
-sleep INF
-`, dpdkResourceName, testpmdCommand, dpdkResourceName, runningTime)},
+			command,
+		},
 		SecurityContext: &corev1.SecurityContext{
 			RunAsUser: pointer.Int64Ptr(0),
 			Capabilities: &corev1.Capabilities{
 				// Enable NET_RAW is required by mellanox nics as they are using the netdevice driver
```

Comment:
can you move the comment above this line (`capabilities := []corev1.Capability{"IPC_LOCK", "SYS_RESOURCE", "NET_RAW"}`)

Reply:
Done